### ISSUE TITLE
Calling a function wrong results in no error

### ISSUE THREAD

**Issue by GiantCrocodile** [TypeStrong/atom-typescript#1208]
Given this code and an empty tsconfig:

```typescript
function stringToUpperCase(s: String) {
  return s.toLocaleUpperCase();
}

stringToUpperCase;
```

`stringToUpperCase` is recognized as the function above and its desciption and parameters are fine. But when I save the file: No errors, all is fine. Even though this function is called wrong bec. of missing paramets, is being recognized as a function and has no brackets it's still valid. Shouldn't be this detected as an issue?

Btw: Is it correct that autocompletion doesn't insert brackets after pressing enter? I'm fine when it doesn't put placeholders for function call but setting the brackets myself annoys me.

**Comment by GiantCrocodile** [comment]
I have no clue if this is still up2date or if this is a real issue. I don't use this package/editor anymore.

**Comment by lierdakil** [comment]
Hi. Thanks for the update.

The bot currently just walks backwards over the backlog, because I don't really have time or motivation to sort this all out (I "inherited" this project with 300+ open issues, most of which have been inapplicable for a while).

Honestly, this is not a "real issue". JS's functions are first-class values, so what you had there wasn't a function call, but an unused value, something akin to
```ts
4;
```
This can (and will) be caught by eslint and tslint.

For the same reason, it doesn't necessarily make sense to add parenthesis after functions, although I can see why this also might be useful, at least in some cases. Thanks for the suggestion, I might implement it as an option at some point.

Anyway, I believe it's ok to close this.

**Comment by GiantCrocodile** [comment]
Thank you for your kind response @lierdakil! Your info makes sense to me now as I wasn't sure who you are when I checked other issues on this repository and I found out something changed compared to back then. Glad to see someone working on this and I wish you the very best with this and on top some very nice holidays 😉.
